Userland/Libraries/LibJS/Runtime/ArrayBuffer.cpp

```diff
@@ -57,6 +57,23 @@ bool ArrayBuffer::is_resizable_array_buffer() const
     return m_max_byte_length.has_value();
 }
 
+// 1.1.7 HostResizeArrayBuffer ( buffer, newByteLength ), https://tc39.es/proposal-resizablearraybuffer/#sec-hostresizearraybuffer
+ThrowCompletionOr<bool> ArrayBuffer::host_resize_array_buffer(GlobalObject& global_object, size_t new_byte_length)
+{
+    // The host-defined abstract operation HostResizeArrayBuffer takes arguments buffer (an ArrayBuffer) and newByteLength (a non-negative integer).
+    // The host-defined abstract operation HostResizeArrayBuffer takes arguments buffer (an ArrayBuffer object) and newByteLength.
+    // It gives the host an opportunity to perform implementation-defined resizing of buffer. If the host chooses not to handle resizing of buffer, it may return unhandled for the default behavior.
+    // The implementation of HostResizeArrayBuffer must conform to the following requirements:
+    //   * The abstract operation must return either NormalCompletion(handled), NormalCompletion(unhandled), or an abrupt throw completion.
+    //   * The abstract operation does not detach buffer.
+    //   * If the abstract operation completes normally with handled, buffer.[[ArrayBufferByteLength]] is newByteLength.
+    // The default implementation of HostResizeArrayBuffer is to return unhandled.
+
+    if (buffer_impl().try_resize(new_byte_length).is_error())
+        return global_object.vm().throw_completion<RangeError>(global_object, ErrorType::NotEnoughMemoryToAllocate, new_byte_length);
+    return true;
+}
```

Comment:
Since this is a Host supplied function it should perhaps be in `JS::VM`, although I can't find any other spec implementing this at the moment so it may be fine to keep it like this for the moment.

Reply:
I think we should do it the right way from the beginning :)

Just follow these existing ones: https://github.com/SerenityOS/serenity/blob/b6c3fad0780faf00802f24797a80b01459f24e2e/Userland/Libraries/LibJS/Runtime/VM.cpp#L49-L114

Reply:
Ah ha! I didn't know about this at all, very cool feature :)